test/typecheck/bad-callref-int32.txt

```diff
@@ -12,4 +12,7 @@
 out/test/typecheck/bad-callref-int32.txt:6:6: error: type mismatch in call_ref, expected reference but got [i32]
     (call_ref (i32.const 10)
      ^^^^^^^^
+out/test/typecheck/bad-callref-int32.txt:6:6: error: type mismatch at end of function, expected [] but got [i32]
+    (call_ref (i32.const 10)
+     ^^^^^^^^
```

Comment:
I'm no quite sure I understand why we now see more errors.  Is this a good thing?

Reply:
I don't think I understand the function-references implementation well enough to give a really considered answer, but I ended up finding a more minimal diff that also matches the current error messages 100%.